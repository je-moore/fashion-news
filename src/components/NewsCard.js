import React from 'react'
import { useNewsCardStyles } from '../styles'
import { Card, CardContent, Typography } from '@material-ui/core'
import { Link } from 'react-router-dom'
import { useHistory } from 'react-router-dom'
import ArticleImage from './ArticleImage'

function NewsCard({ article }) {
  const classes = useNewsCardStyles()

  const history = useHistory()
  const openDialog = id => {
    history.push(`/${id}`)
  }

  return (
    <Card className={classes.card} onClick={() => openDialog(article.id)}>
      <CardContent>
        <ArticleImage imageUrl={article.imageUrl} />
        <Typography
          className={classes.title}
          color="textPrimary"
          variant="h4"
          gutterBottom
        >
          <Link to={`/${article.id}`}>{article.title}</Link>
        </Typography>
      </CardContent>
    </Card>
  )
}

export default NewsCard
