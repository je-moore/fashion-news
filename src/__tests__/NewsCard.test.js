import React from 'react'
import EnzymeToJson from 'enzyme-to-json'
import { mount } from 'enzyme'
import NewsCard from '../components/NewsCard'
import { useHistory } from 'react-router-dom'

jest.mock('react-router-dom', () => {
  return {
    __esModule: true,
    useParams: jest.fn(() => {
      return {
        modalId: 44366,
      }
    }),
    useHistory: jest.fn(() => {
      return {
        length: 50,
        action: 'PUSH',
        location: {
          pathname: '/',
          search: '',
          hash: '',
          key: 'ud1unh',
        },
      }
    }),
    Link: jest.fn(() => null),
  }
})

it('renders correctly', () => {
  const article = {
    title:
      'Interview: Martijn Vlaming van OGER over zijn rol als Personal Advisor bij Italiaans Luxury Maatwerk.',
    url:
      'https://fashionunited.nl/nieuws/mode/interview-martijn-vlaming-van-oger-over-zijn-rol-als-personal-advisor-bij-italiaans-luxury-maatwerk/2019121344367',
    imageUrl:
      '/DlQNIXLzw12YzrwK9k3UutFViomv5EVKEyGRmKDAsTw/gravity:sm/quality:70/aHR0cHM6Ly9zdGF0aWMuZmFzaGlvbnVuaXRlZC5jb20vMjAxOTEyLzRzcXVhcmUtbmluZTEuanBn',
    description:
      'Een carri√®re in de modebranche is een belangrijk onderwerp op FashionUnited. Met deze interviewserie',
    id: 44367,
    __typename: 'LocalNewsArticle',
  }
  const subject = mount(<NewsCard article={article} />)
  expect(EnzymeToJson(subject)).toMatchSnapshot()
})
