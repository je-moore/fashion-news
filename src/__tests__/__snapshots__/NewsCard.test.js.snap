// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<NewsCard
  article={
    Object {
      "__typename": "LocalNewsArticle",
      "description": "Een carriÃ¨re in de modebranche is een belangrijk onderwerp op FashionUnited. Met deze interviewserie",
      "id": 44367,
      "imageUrl": "/DlQNIXLzw12YzrwK9k3UutFViomv5EVKEyGRmKDAsTw/gravity:sm/quality:70/aHR0cHM6Ly9zdGF0aWMuZmFzaGlvbnVuaXRlZC5jb20vMjAxOTEyLzRzcXVhcmUtbmluZTEuanBn",
      "title": "Interview: Martijn Vlaming van OGER over zijn rol als Personal Advisor bij Italiaans Luxury Maatwerk.",
      "url": "https://fashionunited.nl/nieuws/mode/interview-martijn-vlaming-van-oger-over-zijn-rol-als-personal-advisor-bij-italiaans-luxury-maatwerk/2019121344367",
    }
  }
>
  <WithStyles(ForwardRef(Card))
    className="makeStyles-card-1"
    onClick={[Function]}
  >
    <ForwardRef(Card)
      className="makeStyles-card-1"
      classes={
        Object {
          "root": "MuiCard-root",
        }
      }
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(Paper))
        className="MuiCard-root makeStyles-card-1"
        elevation={1}
        onClick={[Function]}
      >
        <ForwardRef(Paper)
          className="MuiCard-root makeStyles-card-1"
          classes={
            Object {
              "elevation0": "MuiPaper-elevation0",
              "elevation1": "MuiPaper-elevation1",
              "elevation10": "MuiPaper-elevation10",
              "elevation11": "MuiPaper-elevation11",
              "elevation12": "MuiPaper-elevation12",
              "elevation13": "MuiPaper-elevation13",
              "elevation14": "MuiPaper-elevation14",
              "elevation15": "MuiPaper-elevation15",
              "elevation16": "MuiPaper-elevation16",
              "elevation17": "MuiPaper-elevation17",
              "elevation18": "MuiPaper-elevation18",
              "elevation19": "MuiPaper-elevation19",
              "elevation2": "MuiPaper-elevation2",
              "elevation20": "MuiPaper-elevation20",
              "elevation21": "MuiPaper-elevation21",
              "elevation22": "MuiPaper-elevation22",
              "elevation23": "MuiPaper-elevation23",
              "elevation24": "MuiPaper-elevation24",
              "elevation3": "MuiPaper-elevation3",
              "elevation4": "MuiPaper-elevation4",
              "elevation5": "MuiPaper-elevation5",
              "elevation6": "MuiPaper-elevation6",
              "elevation7": "MuiPaper-elevation7",
              "elevation8": "MuiPaper-elevation8",
              "elevation9": "MuiPaper-elevation9",
              "root": "MuiPaper-root",
              "rounded": "MuiPaper-rounded",
            }
          }
          elevation={1}
          onClick={[Function]}
        >
          <div
            className="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-1 MuiPaper-rounded"
            onClick={[Function]}
          >
            <WithStyles(ForwardRef(CardContent))>
              <ForwardRef(CardContent)
                classes={
                  Object {
                    "root": "MuiCardContent-root",
                  }
                }
              >
                <div
                  className="MuiCardContent-root"
                >
                  <ArticleImage
                    imageUrl="/DlQNIXLzw12YzrwK9k3UutFViomv5EVKEyGRmKDAsTw/gravity:sm/quality:70/aHR0cHM6Ly9zdGF0aWMuZmFzaGlvbnVuaXRlZC5jb20vMjAxOTEyLzRzcXVhcmUtbmluZTEuanBn"
                  >
                    <img
                      alt=""
                      className="makeStyles-articleImg-32"
                      src="https://fashionunited.com/DlQNIXLzw12YzrwK9k3UutFViomv5EVKEyGRmKDAsTw/gravity:sm/quality:70/aHR0cHM6Ly9zdGF0aWMuZmFzaGlvbnVuaXRlZC5jb20vMjAxOTEyLzRzcXVhcmUtbmluZTEuanBn"
                    />
                  </ArticleImage>
                  <WithStyles(ForwardRef(Typography))
                    className="makeStyles-title-2"
                    color="textPrimary"
                    gutterBottom={true}
                    variant="h4"
                  >
                    <ForwardRef(Typography)
                      className="makeStyles-title-2"
                      classes={
                        Object {
                          "alignCenter": "MuiTypography-alignCenter",
                          "alignJustify": "MuiTypography-alignJustify",
                          "alignLeft": "MuiTypography-alignLeft",
                          "alignRight": "MuiTypography-alignRight",
                          "body1": "MuiTypography-body1",
                          "body2": "MuiTypography-body2",
                          "button": "MuiTypography-button",
                          "caption": "MuiTypography-caption",
                          "colorError": "MuiTypography-colorError",
                          "colorInherit": "MuiTypography-colorInherit",
                          "colorPrimary": "MuiTypography-colorPrimary",
                          "colorSecondary": "MuiTypography-colorSecondary",
                          "colorTextPrimary": "MuiTypography-colorTextPrimary",
                          "colorTextSecondary": "MuiTypography-colorTextSecondary",
                          "displayBlock": "MuiTypography-displayBlock",
                          "displayInline": "MuiTypography-displayInline",
                          "gutterBottom": "MuiTypography-gutterBottom",
                          "h1": "MuiTypography-h1",
                          "h2": "MuiTypography-h2",
                          "h3": "MuiTypography-h3",
                          "h4": "MuiTypography-h4",
                          "h5": "MuiTypography-h5",
                          "h6": "MuiTypography-h6",
                          "noWrap": "MuiTypography-noWrap",
                          "overline": "MuiTypography-overline",
                          "paragraph": "MuiTypography-paragraph",
                          "root": "MuiTypography-root",
                          "srOnly": "MuiTypography-srOnly",
                          "subtitle1": "MuiTypography-subtitle1",
                          "subtitle2": "MuiTypography-subtitle2",
                        }
                      }
                      color="textPrimary"
                      gutterBottom={true}
                      variant="h4"
                    >
                      <h4
                        className="MuiTypography-root makeStyles-title-2 MuiTypography-h4 MuiTypography-colorTextPrimary MuiTypography-gutterBottom"
                      >
                        Interview: Martijn Vlaming van OGER over zijn rol als Personal Advisor bij Italiaans Luxury Maatwerk.
                      </h4>
                    </ForwardRef(Typography)>
                  </WithStyles(ForwardRef(Typography))>
                </div>
              </ForwardRef(CardContent)>
            </WithStyles(ForwardRef(CardContent))>
          </div>
        </ForwardRef(Paper)>
      </WithStyles(ForwardRef(Paper))>
    </ForwardRef(Card)>
  </WithStyles(ForwardRef(Card))>
</NewsCard>
`;
